src/core/aws-net-websocket-registry.adb

```diff
@@ -568,6 +572,24 @@ package body AWS.Net.WebSocket.Registry is
          Registered.Clear;
       end Finalize;
 
+      ----------
+      -- Free --
+      ----------
+
+      procedure Free (WebSocket : in out Object_Class) is
+      begin
+         --  If WebSocket is in Sending it means that it has been
+         --  taken by the Get_Socket call. We cannot free it now, we
+         --  record this socket to be freed as soon as it is released
+         --  (Release_Socket) call.
+
+         if Sending.Contains (WebSocket.Id) then
```

Comment:
I was looking in the code why Sending is a set. It seems to be it might just as well be a flag in the Object itself, rather than an independent set. The difficulty is to make sure it is only accessed via the protected object (though it could also be made atomic for instance).

Reply:
Yes, maybe but could be done as a clean-up and does not seem do be the issue at hand.